optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/PlannerBenchmarkRunner.java

```diff
@@ -122,54 +124,94 @@ private void initBenchmarkDirectoryAndSubdirs() {
     }
 
     private void warmUp() {
-        if (plannerBenchmarkResult.getWarmUpTimeMillisSpentLimit() > 0L) {
-            logger.info("================================================================================");
-            logger.info("Warm up started");
-            logger.info("================================================================================");
-            long startingTimeMillis = System.currentTimeMillis();
-            long timeLeft = plannerBenchmarkResult.getWarmUpTimeMillisSpentLimit();
-            List<ProblemBenchmarkResult> unifiedProblemBenchmarkResultList = plannerBenchmarkResult.getUnifiedProblemBenchmarkResultList();
-            Iterator<ProblemBenchmarkResult> it = unifiedProblemBenchmarkResultList.iterator();
-            while (timeLeft > 0L) {
-                if (!it.hasNext()) {
-                    it = unifiedProblemBenchmarkResultList.iterator();
-                }
-                ProblemBenchmarkResult problemBenchmarkResult = it.next();
-                timeLeft = warmUp(problemBenchmarkResult, startingTimeMillis, plannerBenchmarkResult.getWarmUpTimeMillisSpentLimit(), timeLeft);
-            }
-            logger.info("================================================================================");
-            logger.info("Warm up ended");
-            logger.info("================================================================================");
+        if (plannerBenchmarkResult.getWarmUpTimeMillisSpentLimit() <= 0L) {
+            return;
+        }
+        logger.info("================================================================================");
+        logger.info("Warm up started");
+        logger.info("================================================================================");
+        long timeLeft = plannerBenchmarkResult.getWarmUpTimeMillisSpentLimit();
+        long timeEnd = System.currentTimeMillis() + timeLeft;
+        Map<SingleBenchmarkRunner, Future<SingleBenchmarkRunner>> futureMap
+                = new HashMap<SingleBenchmarkRunner, Future<SingleBenchmarkRunner>>();
+        Map<SolverBenchmarkResult, TerminationConfig> originalTerminationConfigMap
+                = new HashMap<SolverBenchmarkResult, TerminationConfig>();
+        while (timeLeft > 0L) {
```

Comment:
Suppose we have a warmup of 400 seconds. Suppose we have a parallel benchmark count of 4. Suppose we have 10 solver configurations. How long does each warmup single benchmark get?

The philosophy is that each solver config needs to be warmed up (but not every problem dataset), but that the user only specifies the entire budget (usually 30 seconds).
In the old code, so with parallel count 1, if we have 400 seconds with 10 solver configurations, each solver config would get 40 seconds. If the first one only took 13 seconds (instead of 40), then the other 9 would have gotten (400-13)/9 = 43 seconds. Basically, any time not soaked up by previous solver configuration runs, is donated to later runs.